packages/hash/frontend/src/components/hooks/useAuthenticatedUser.ts

```diff
@@ -120,6 +132,11 @@ export const useAuthenticatedUser = (
   //   client.writeQuery();
   // };
 
+  if (!prevAuthenticatedUser && authenticatedUser) {
+    setPrevAuthenticatedUser(authenticatedUser);
+    void onAuthenticate?.(authenticatedUser);
```

Comment:
I think this needs discussion on Tandem. It's possibly okay, but there are some concerns worth discussing. It may be surprising to consumers of this hook that `onAuthenticate` fires during render.